pkg/gardenlet/controller/shoot/shoot/reconciler_delete.go

```diff
@@ -259,7 +258,7 @@ func (r *Reconciler) runDeleteShootFlow(ctx context.Context, o *operation.Operat
 				waitUntilEtcdReady,
 				waitUntilKubeAPIServerServiceIsReady,
 				waitUntilControlPlaneReady,
-			).InsertIf(!staticNodesCIDR),
+			),
```

Comment:
Could you please share why you think that this is dead code?

Reply:
"dead code" may not be the optimal term, however, `InsertIf(condition bool, iders ...TaskIDer)` https://github.com/gardener/gardener/blob/fc31b86d6da5a30d87a9e62bbd61ff9aaf0c9de0/pkg/utils/flow/taskid.go#L66 with an empty `iders` list does not modify the TaskIDs set. That is, the call to `InsertIf` has no effect and can simply be removed.

Reply:
Good catch.

Indeed this code does not seem to cause any effect regardless of the condition. Hence, removing it after over three years seems like a good choice.